File: internal/domain/session.go
package domain

import "time"

type Session struct {
	ID           string            `json:"id"`
	UserID       int64             `json:"user_id"`
	Token        string            `json:"token"`         //current access token
	RefreshToken string            `json:"refresh_token"` // refresh token
	DeviceInfo   map[string]string `json:"device_info"`
	IPAddress    string            `json:"ip_address"`
	UserAgent    string            `json:"user_agent"`
	CreatedAt    time.Time         `json:"created_at"`
	LastActive   time.Time         `json:"last_active"`
	IsActive     bool              `json:"is_active"`
}

File: frontend/src/types/models/session.ts
import type { UserDevice } from './user_device'

export type SessionResponse = {
	sessions: Session[]
}

export type Session = {
	user_id: number
	token: string
	device_info: UserDevice
	ip_address: string
	user_agent: string
	created_at: string
	last_active: string
	is_active: boolean
}
